### PR TITLE
Test using client template

### PR DESCRIPTION
**Pull request by weshaggard**
The work @mitchdenny did to group the artifacts in https://github.com/Azure/azure-sdk-for-java/pull/13511/files didn't account for the track 1 libraries so they are currently broken for release.

Looks like https://github.com/Azure/azure-sdk-for-java/blob/master/eng/pipelines/templates/jobs/archetype-sdk-client.yml (track 2 template) and https://github.com/Azure/azure-sdk-for-java/blob/master/eng/pipelines/templates/jobs/archetype-sdk-data.yml (track 1 template) have diverged a fair amount since they were split. This PR is attempting to switch one of the track 1 pipelines, eventhubs which @conniey is currently trying to release, over to just use the client template. 

Assuming we can get this to work for eventhubs we should remove the data template completely and move all the other track 1 libraries to use the client template as well.

### CODE REVIEW

File: eng/pipelines/templates/jobs/archetype-sdk-client.yml [hunk 1]
```diff
@@ -167,12 +167,13 @@ jobs:
           Directory: sdk/${{ parameters.ServiceDirectory }}
           CheckLinkGuidance: $true
 
-      - ${{ each artifact in parameters.Artifacts }}:
-        - template: /eng/common/pipelines/templates/steps/verify-changelog.yml
-          parameters:
-            PackageName: ${{artifact.name}}
-            ServiceName: ${{parameters.ServiceDirectory}}
-            ForRelease: false
+      - ${{if ne(parameters.SDKType, 'data')}}:
```
Comment (weshaggard) [inline eng/pipelines/templates/jobs/archetype-sdk-client.yml:170]: A lot of track 1 libraries don't have change logs and we already shipped those checks in the release steps so I'm skipping them here as well.

File: eng/pipelines/templates/jobs/archetype-sdk-client.yml [hunk 2]
```diff
@@ -224,7 +225,12 @@ jobs:
       - template: ../variables/globals.yml
 
     strategy:
-      matrix: ${{ parameters.TestMatrix }}
+      matrix:
+        ${{ each matrixEntry in parameters.TestMatrix }}:
+          # Skip TestFromSource jobs for SDKType data
+          ${{ if not(and(eq(matrixEntry.value.TestFromSource, 'true'), eq(parameters.SDKType, 'data'))) }}:
```
Comment (weshaggard) [inline eng/pipelines/templates/jobs/archetype-sdk-client.yml:231]: TestFromSource will be difficult to get working for track 1 libraries and I'm not sure it is worth the efforts so skipping that test entry.

**Review by conniey — APPROVED** [review]
🥳

**Comment by weshaggard** [comment]
I filed https://github.com/Azure/azure-sdk-for-java/issues/16391 to track making the switch for the other track 1 libraries.

**Comment by weshaggard** [comment]
Merging now as the last job isn't going to add any validation.

**Pull request merged**